pkg/apis/config/defaults.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+const (
+	// DefaultsConfigName is the name of config map for the defaults.
+	DefaultsConfigName = "config-defaults"
+)
+
+// NewDefaultsConfigFromMap creates a Defaults from the supplied Map
+func NewDefaultsConfigFromMap(data map[string]string) (*Defaults, error) {
+	nc := &Defaults{}
+
+	// Process int64 fields
+	for _, i64 := range []struct {
+		key   string
+		field *int64
+		// specified exactly when optional
+		defaultValue int64
+	}{{
+		key:          "revision-timeout-seconds",
+		field:        &nc.RevisionTimeoutSeconds,
+		defaultValue: DefaultRevisionTimeoutSeconds,
+	}} {
+		if raw, ok := data[i64.key]; !ok {
+			*i64.field = i64.defaultValue
+		} else if val, err := strconv.ParseInt(raw, 10, 64); err != nil {
```

Comment:
Why not string and parse duration?

Reply:
It isn't a duration, we made the field seconds explicitly.